mitodl/bootcamp-ecommerce
# Front-end code for profiles, registration

#### What are the relevant tickets?
Closes #398
Closes #451
Closes #379 

#### What's this PR do?
Adds frontend code for profile pages and auth functionality, copied over from xpro, and installs required packages.  It does not add any links/buttons to the home page to get to the login, registration, or profile pages - that will be in a future PR.

#### How should this be manually tested?
With the setting `FEATURE_SOCIAL_AUTH_API` set to `True` or `False`, all the prior functionality (front page, login/registration via edx, payment page, receipt page) should still work. Tests should all pass.

- Set `FEATURE_SOCIAL_AUTH_API=True`
- Start at `http://bc.odl.local:8099/create-account/` and go through the registration flow.  You should get sent a confirmation email (if mailgun settings are properly configured), the confirmation link should send you to step 1 then 2 of new account creation, and finally to the payment page.
- Sign out.  Then go to `http://bc.odl.local:8099/signin/` and sign in as the new user.  You should be asked for your password then sent to the payment page.
- Go to `http://bc.odl.local:8099/account-settings/` and change your email.  You should get a confirmation link.  Clicking the link should bring you back to the account settings page and show your new email.
- Change your password.  Log out then back in with the new password.
- Once logged in, go to `../profile` and you should see your profile info.  Click the 'Edit Profile' button.  The profile edit form should load.  Make some changes and save, you should return to the profile page and it should reflect the changes you made.

#### Screenshots

<img width="849" alt="Screen Shot 2020-05-01 at 7 04 25 AM" src="https://user-images.githubusercontent.com/187676/80801415-5665bb00-8b7a-11ea-8ebe-cd1abe4a879e.png">

<img width="901" alt="Screen Shot 2020-05-01 at 7 04 49 AM" src="https://user-images.githubusercontent.com/187676/80801413-52399d80-8b7a-11ea-9d29-dc1f03c4adeb.png">
<img width="861" alt="Screen Shot 2020-05-01 at 7 03 52 AM" src="https://user-images.githubusercontent.com/187676/80801428-62ea1380-8b7a-11ea-9666-6b6b3743363f.png">


<img width="936" alt="Screen Shot 2020-05-04 at 4 34 30 PM" src="https://user-images.githubusercontent.com/187676/81012521-acab5600-8e27-11ea-8e14-7d9e68c44523.png">

<img width="935" alt="Screen Shot 2020-05-04 at 4 35 27 PM" src="https://user-images.githubusercontent.com/187676/81012534-b03edd00-8e27-11ea-85df-d841f6ae2a52.png">

<img width="817" alt="Screen Shot 2020-05-04 at 5 00 18 PM" src="https://user-images.githubusercontent.com/187676/81013222-c8632c00-8e28-11ea-906f-c126b496f237.png">


<img width="997" alt="Screen Shot 2020-05-04 at 5 01 56 PM" src="https://user-images.githubusercontent.com/187676/81013425-1710c600-8e29-11ea-9b60-1b6adf4aacdc.png">

<img width="1039" alt="Screen Shot 2020-05-04 at 5 02 33 PM" src="https://user-images.githubusercontent.com/187676/81013433-1b3ce380-8e29-11ea-9dec-12ba7dd5cd0e.png">
